### ISSUE TITLE
LibreLinkUp issues?

### ISSUE THREAD

**Issue by zanfirovidius** [timoschlueter/nightscout-librelink-up#137]
Hello. Do any of you have issues with LibreLinkUp? It will receive (or not process) data from Libre 3 any 30-ish minutes or so. And because of this Nighscout will receive data in the same way.

I am using FR region.
And this started to happen around 6 May 2024.

**Comment by Zurnl** [comment]
Hey, i'm having the same problem with Libre3. 
It started yesterday, 8th May 2024. I am using EU region configured to 1min. It gets reading randomly between 1-20 minutes apart. 
Changed region from EU to FI, where i live but this does not have any effect.

**Comment by wklimowicz** [comment]
Same here. Latest versions of Nightscout and LibreLinkUp on docker on Northflank. Attached the typical log (1 successful download, followed by a few failures -- my refresh rate is set to 5m and seems to work randomly <50% of the time). The LibreLinkUp app seems to be working fine in real time.

<details>
  <summary>Northflank LibreLinkUp Logs</summary>
  
```
2024-05-09T21:20:00.752893968Z stdout F [error]: getting libreLinkUpConnection:  Cannot read properties of undefined (reading 'data')
2024-05-09T21:20:00.751918625Z stdout F [error]: {}
2024-05-09T21:15:00.820913864Z stdout F [info]: Upload of 2 measurements to Nightscout succeeded
2024-05-09T21:15:00.713320983Z stdout F [info]: Trying to upload 2 glucose measurement items to Nightscout
2024-05-09T21:15:00.566991959Z stdout F [info]: -> The following connection will be used: [[My details]]
2024-05-09T21:15:00.566913382Z stdout F [info]: Found 1 LibreLink Up connection.
2024-05-09T21:15:00.480990798Z stdout F [info]: Logged in to LibreLink Up
2024-05-09T21:15:00.284073143Z stdout F [info]: renew token
2024-05-09T21:10:01.193120315Z stdout F [error]: getting libreLinkUpConnection:  Cannot read properties of undefined (reading 'data')
2024-05-09T21:10:01.19243358Z stdout F [error]: {}
2024-05-09T21:10:01.156515482Z stdout F [info]: Logged in to LibreLink Up
2024-05-09T21:10:00.910913795Z stdout F [info]: renew token
2024-05-09T21:05:00.798435294Z stdout F [error]: getting libreLinkUpConnection:  Cannot read properties of undefined (reading 'data')
2024-05-09T21:05:00.798130477Z stdout F [error]: {}
2024-05-09T21:05:00.734735873Z stdout F [info]: Logged in to LibreLink Up
2024-05-09T21:05:00.53635479Z stdout F [info]: renew token
2024-05-09T21:00:00.492920215Z stdout F [error]: getting libreLinkUpConnection:  Cannot read properties of undefined (reading 'data')
2024-05-09T21:00:00.492449676Z stdout F [error]: {}
2024-05-09T21:00:00.438962283Z stdout F [info]: Logged in to LibreLink Up
2024-05-09T21:00:00.175678596Z stdout F [info]: renew token
```
</details>

**Comment by zanfirovidius** [comment]
While I was testing this issue I noticed that LibreLinkUp will also update its value "randomly"> I have installed LibreLinkUp on another iPhone to check if the issue is between LibreLinkUp services and my nightscout-librelink-up server or between nightscout-librelink-up server and my Nightscout instance that I am running on Heroku. 

I hope that Freestyle did not decide to make some changes/implement some restriction that will prevent the use of 3rd-party solutions like nightscout-librelink-up

**Comment by Zurnl** [comment]
Just updated LibreLinkUp to 2.6.0. So far LibreLinkUp works without the errors explained before the update.
Changed region back to EU because version 2.6.0 does not allow FI region. I'm surprised it worked before the update.
Big thanks to developers!

**Comment by timoschlueter** [comment]
@Zurnl: Great to hear that it works for you now! :)

I just want to give a little inside into the "Region-Situation": Prior to 2.6.0 you could set anything you wanted as the region variable. If the region would actually be supported, the corresponding LLU endpoint would be used to retrieve the data. Since "FI" is no region with a dedicated endpoint provided by Abbott, all versions <2.6.0 would automatically fall back to the "EU" region. So if you set the region to "FI" it actually automatically used "EU" since that has been the fallback so far.

With 2.6.0 we made the region setting more explicit, since many users ran into problems with it in the past. Since there is no dedicated FI-Region on Abbotts side yet, you are are totally fine using the EU region :)

**Comment by timoschlueter** [comment]
@zanfirovidius have you also tried v2.6.0 yet? We added a lot of improvements to make the connection to Abbotts servers more reliably.

**Comment by wklimowicz** [comment]
The issue seems to have fixed itself on it's own midway through May 14th on the old version (2.5.1 I think). I've now upgraded to 2.6.0 and no issues so far. Thanks for your work maintaining this!

**Comment by zanfirovidius** [comment]
@timoschlueter I will try tonight to do the update and see if it will fix the issue. Thank you.

**Comment by zanfirovidius** [comment]
sorry to say but it is still the same. i will investigate more on this.

**Comment by ezepizarro** [comment]
If it's not related please disregard, but I have the same issue and I am running latest versions.
AAPS don't update for hours. I tested all conections and they are good.

**Comment by mdavidsen** [comment]
Seems this is something Abbott is constantly doing apparently. 
Lost access via Linode in Stockholm, with the same error as above. Now I've running this at at home, with nightscout still hosted at Linode. 

I had the cron-job set to `*/2.5 * * * *`, I'll consider lowering it. But coming from libre2, it just feels wrong to have that much latency extra for Loop. 

 So much for owning your own health data. This is for the NO-region.

**Comment by timoschlueter** [comment]
I have to agree with @mdavidsen. There is nothing we can do about this unfortunately. :(
Increasing the interval is all that we can do.
